wbooks/src/interfaces/rootState.ts

```diff
@@ -1,3 +1,8 @@
-import { rootReducer } from '@redux/store';
+import { $CombinedState } from 'redux';
+import { BookState } from '@interfaces/bookState';
 
-export type RootState = ReturnType<typeof rootReducer>;
+export type RootState = {
+  readonly [$CombinedState]?: undefined;
+} & {
+  books: BookState;
+};
```

Comment:
Why all of this complex logic for an interface?
```
export interface RootState = {
  books: BookState;
}
```

Reply:
You're right, I thought we had to consider that all reducers are wrapped with redux's `combineReducers` function, apparently not the case, good to know.